pkg/bitcoin/transaction_test.go

```diff
@@ -72,6 +72,72 @@ func TestTransaction_SerializeRoundtrip(t *testing.T) {
 	}
 }
 
+func TestTransaction_SerializeVersion(t *testing.T) {
+	transaction := transactionFixture(t)
+
+	serializedVersion := transaction.SerializeVersion()
+
+	testutils.AssertBytesEqual(
+		t,
+		hexToSlice(t, "01000000"),
+		serializedVersion[:],
+	)
+}
+
+func TestTransaction_SerializeInputs(t *testing.T) {
+	transaction := transactionFixture(t)
+
+	serializedInputs := transaction.SerializeInputs()
+	expectedSerializedInputs := hexToSlice(
+		t,
+		"036896f9abcac13ce6bd2b80d125bedf997ff6330e999f2f60"+
+			"5ea15ea542f2eaf80000000000ffffffffed0ae94da996c6f3b89dfe967675d"+
```

Comment:
Why is the transaction hash reversed when serialized? I am looking at `transactionFixture` and the outpoint hash of `f8eaf242a55ea15e602f9f990e33f67f99dfbe25d1802bbde63cc1caabf99668`. It is present in the serialized version but the byte order is changed.

Reply:
The serialized hash must be in the `bitcoin.InternalByteOrder`. The `transactionFixture` takes a hash in the human-friendly `bitcoin.ReversedByteOrder` and uses `hexToHash` to construct a proper `bitcoin.Hash` instance. We pass hashes in `bitcoin.ReversedByteOrder` there as it is easier to check them in block explorers out of the box without any conversion.

Reply:
Can we add your answer to my question as a comment to `transactionFixture`?

Reply:
Sure 94d72d77